### PR TITLE
[WIP/MIG][9.0] Migrate magentoerpconnect

### PR DESCRIPTION
**Pull request by SodexisTeam**
Needs label : **Work in progress**

Changes done:
- Obtained the histories from 8.0 using https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0
- Checked pep8 conventions using oca-autopep8
- Version and dependency change
- Changed to simple headers
- Modified README.rst
- Add import for virtual and downloadable products as service and digital respectively from PR https://github.com/OCA/connector-magento/pull/214

### CODE REVIEW

File: magentoerpconnect/product_category.py [hunk 1]
```diff
@@ -265,7 +250,7 @@ def parent_id(self, record):
         mag_cat_id = binder.to_openerp(record['parent_id'])
 
         if category_id is None:
-            raise MappingError("The product category with "
-                               "magento id %s is not imported." %
+            raise MappingError(_("The product category with "
+                                 "magento id %s is not imported.") %
```
Comment (guewen) [inline magentoerpconnect/product_category.py]: Those are technical errors, not meant to be translated (same comment at the other places).

**Comment by guewen** [comment]
Could be nice to use https://github.com/kevin1024/vcrpy for the test data.

**Comment by guewen** [comment]
Exceptions of type `UserError` are shown to the end user and should be translated. I'm really not fond of translatable error messages for the low-level errors such as the network or mapping errors.

**Comment by PhilippeBach** [comment]
a little suggest for the V9   for categories could we import the categories in the same fields than odoo Ecommerce? If one time you want to develop odoo frontend  you dont need to make them one time more 

my post could be a a wrong place but there is no real board for discutions and suggest for OCA 

Regards 

Philippe

File: magentoerpconnect/product_category.py (hunk 1 above)
Comment (atchuthan) [inline magentoerpconnect/product_category.py]: We have similar errors in community and odoo modules that contain translation _(...) for technical errors.

Comment (guewen) [inline magentoerpconnect/product_category.py]: What I mean by technical errors is that they are not supposed to be shown to the users with a `UserError` popup.

I distinguish "error messages" with are meant entirely for the users and should be localized and "exceptions messages" that must not be translated. These exceptions messages are for developers and not end-users. This is important to be able to search for errors and in some case to analyze the content of the exception in `try: except:`.

Comment (atchuthan) [inline magentoerpconnect/product_category.py]: thanks for your info... will remove translation for technical errors

Comment (atchuthan) [inline magentoerpconnect/product_category.py]: For info, in the lint check for Travis, it tests for error raised with translations.

magentoerpconnect/sale.py:359: [C8107(translation-required), SaleImportRule._rule_authorized] String parameter of raise "OrderImportRuleRetry" requires translation. Use _('The order has not been authorized.\nThe import will be retried later.')

Gonna ignore this case for technical errors.

**Comment by StefanRijnhart** [comment]
What is the status of this? The commit history and the diff looks mangled, with commits by transifex etc.

**Comment by atchuthan** [comment]
@StefanRijnhart rebase done

**Comment by StefanRijnhart** [comment]
Thanks for the rebase. Can you check the output of the travis jobs?

**Comment by berpweb** [comment]
Any updates with this PR? I restarted the runbot which looks fine now. Travis 👎

**Comment by berpweb** [comment]
Someone working on this or just an abandoned PR?
Ping: @SodexisTeam 

@guewen #255 Is this PR supersede the current one here for V9?

**Comment by guewen** [comment]
@berpweb I'm working on the v10.0 on #255.

**Pull request closed without merging**